storage/src/main/java/zipkin2/storage/kafka/KafkaStorage.java

```diff
@@ -81,8 +86,9 @@
   volatile AdminClient adminClient;
   volatile Producer<String, byte[]> producer;
   volatile KafkaStreams traceAggregationStream, traceStoreStream, dependencyStoreStream;
-  volatile boolean closeCalled, topicsValidated;
-  long minTracesStored;
+  volatile Server server;
```

Comment:
I think it is better to re-use the existing armeria server if the communication is http. just use another base path.

For example, gRPC, we mount at another root. This way you don't need to worry about metrics either at least not http metrics.

Right now `ArmeriaServerConfigurator` is actually in the spring package.. this may be ok because it allows it to do spring things such as Ordered etc, however I don't think it is a good idea to use spring dep. Until/unless there's a base type probably accept `Consumer<ServerBuilder>` to configure the paths you want to expose

cc @minwoox

Reply:
`Consumer<ServerBuilder>` is a great way to configure server without deps :)

And agree it should be much better to configure the existing server.

Reply:
+1 on sharing communication. Added `consumer<serverBuilder>` as well; looking forward to when armeria loads service based on it :)